### DISCUSSION TITLE
CSV: column separator not recognized

### DISCUSSION THREAD

**Discussion by Pierre-de-la-Verre** [WinMerge/winmerge#696] [Q&A]
[xxx.zip](https://github.com/WinMerge/winmerge/files/6206127/xxx.zip)

I have two CSV with "semicolon ;" as separator, but when I want to compare it does not use the separator and puts everything in the first column. Why?
![grafik](https://user-images.githubusercontent.com/13277944/112503609-5a4e8d00-8d8b-11eb-9667-1623d0d6fdf0.png)

But a "comma ," is used in this case instead of the semicolon

**Comment by sdottaka** [comment]
The files have a csv extension, so it is considered "Comma-separated-values-file".

Unfortunately, WinMerge doesn't have a good way to handle files that aren't comma-separated-values, even though they have a csv extension.

As a workaround, set the following, change the file extension to ssv, and compare the files.
![image](https://user-images.githubusercontent.com/98126/112557974-08504a80-8e11-11eb-81f8-ab3e5c6281e1.png)

**Comment by Pierre-de-la-Verre** [comment]
Thanks. I see that I misunderstood the options dialogue. OK for me.

**Comment by CSBVision** [comment]
I also ran into this issue. Another option seems to be to simply delete '*.csv' in the csv-file pattern box and add  '*.csv' to the custom delimiter box with the separator of choice, e.g. ';'.
Would be great if the csv-delimiter could be selected directly, but this workaround seems to work better than renaming files if csv-files are usually separated by semicolons.

**Reply by srgloureiro** [reply]
I second @CSBVision's opinion!

![image](https://user-images.githubusercontent.com/6369967/169153145-697ada07-4a98-489e-a95e-3500fb8211e4.png)
